websites/C/Cosmote TV/presence.ts

```diff
@@ -5,108 +5,193 @@ const presence = new Presence({
+			},
+			"onDemand/FILMS": {
+				details: "Browsing Films",
+			},
+			"onDemand/SERIES": {
+				details: "Browsing Series",
+			},
+			"onDemand/MOVIES_CLUB": {
+				details: "Browsing Movies Club",
+			},
+			"onDemand/SPORTS": {
+				details: "Browsing Sports",
+			},
+			"onDemand/DOCUMENTARIES": {
+				details: "Browsing Documentaries",
+			},
+			"onDemand/KIDS": {
+				details: "Browsing Kids Content",
+			},
+		},
+		[logo, timestamps] = await Promise.all([
+			presence.getSetting<boolean>("logo"),
+			presence.getSetting<boolean>("timestamps"),
+		]);
+
+	for (const [path, data] of Object.entries(pages)) {
+		if (document.location.hash.includes(path))
+			presenceData = { ...presenceData, ...data };
+	}
+
+	if (document.querySelector<HTMLDivElement>("div[ng-if='showPlayer']")) {
+		const { paused, currentTime, duration } =
+			document.querySelector<HTMLVideoElement>("video#arxPlayer");
+		// TV
+		if (
+			document.querySelector<HTMLImageElement>(
+				"div[ng-if='details.channelLogoWide'] > .wide-logo"
+			)
+		) {
+			presenceData.details = document.querySelector<HTMLSpanElement>(
 				".meta-title[ng-bind='details.title']"
-			)?.textContent,
-			{ paused, currentTime, duration } =
-				document.querySelector<HTMLVideoElement>("video#arxPlayer"),
-			live = document.querySelector(".meta-remain") ? true : false;
+			)?.textContent;
+			presenceData.state = document.querySelector<HTMLSpanElement>(
+				".meta-title[ng-bind='details.channel.title']"
+			)?.textContent;
+			if (logo) {
+				presenceData.largeImageKey = document
+					.querySelector<HTMLImageElement>(
+						"div[ng-if='details.channelLogoWide'] > .wide-logo"
+					)
+					?.src.replace("-wide", "-normal");
+			}
 
-		if (!live) {
-			presenceData.smallImageKey = paused ? "pause" : "play";
-			presenceData.smallImageText = paused
-				? (await strings).pause
-				: (await strings).play;
+			// Live
+			if (document.querySelector<HTMLSpanElement>(".meta-remain")) {
+				if (channel !== presenceData.state || !channelTimestamp)
+					channelTimestamp = Math.floor(Date.now() / 1000);
 
-			presenceData.endTimestamp = presence
-				.getTimestamps(Math.floor(currentTime), Math.floor(duration))
-				.pop();
+				presenceData.startTimestamp = channelTimestamp;
 
+				presenceData.smallImageKey = paused ? "pause" : "live";
+				presenceData.smallImageText = paused
+					? (await strings).pause
+					: (await strings).live;
+			} else {
+				// Replay / Timeshift
+				[presenceData.startTimestamp, presenceData.endTimestamp] =
+					presence.getTimestamps(
+						presence.timestampFromFormat(
+							document.querySelector<HTMLSpanElement>("#VcurrentTime")
+								.textContent
+						),
+						presence.timestampFromFormat(
+							document.querySelector<HTMLSpanElement>("#Vduration").textContent
+						)
+					);
+
+				presenceData.smallImageKey = paused ? "pause" : "play";
+				presenceData.smallImageText = paused
+					? (await strings).pause
+					: (await strings).play;
```

Comment:
await the strings at the top of the file instead. @EncryptedDev for more info.

Reply:
Like thishttps://github.com/PreMiD/Presences/blob/b622198bdf2396867a11ed644a3725d06e2a6745/websites/S/Skribbl.io/presence.ts#L14